sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/NettyToAzureCoreHttpHeadersWrapper.java

```diff
@@ -92,15 +92,12 @@ public HttpHeaders add(String name, String value) {
 
     @Override
     public HttpHeader get(String name) {
-        if (nettyHeaders.contains(name)) {
-            // Be careful here: Netty's HttpHeaders 'get' method will return only the first value,
-            // which is obviously not what we want to call!
-            // We call 'getAll' instead, but unfortunately there is a representation mismatch:
-            // Netty HttpHeaders uses List<String>, whereas azure-core HttpHeaders joins it all into a
-            // comma-separated String.
-            return new NettyHttpHeader(this, name, nettyHeaders.getAll(name));
-        }
-        return null;
+        // Be careful here: Netty's HttpHeaders 'get' method will return only the first value, which is obviously not
+        // what we want to call! We call 'getAll' instead, but unfortunately there is a representation mismatch:
+        // Netty HttpHeaders uses List<String>, whereas azure-core HttpHeaders joins it all into a comma-separated
+        // String. Additionally, 'getAll' will return an empty list if there is no value(s) for the header.
+        List<String> values = nettyHeaders.getAll(name);
```

Comment:
We should check that it is faster to 'get values and null check', versus 'contains check first'.

Reply:
It should be faster when the header exists (without running formal validation). 

Internally, from what source code I read, Netty uses arrays that act as a direct Map bucket to contain headers where each bucket is a linked list, or a node list. `contains` and `getAll` both use the header name to determine which bucket may contain that header, if it exists, but `contains` will fast return true if a header with a matching name is found where `getAll` will iterate all headers in the bucket to find all values. Based on this, `getAll` without contains should be faster as we remove the `contains` bucket determination and any potential node looping by just doing a single loop on the nodes getting all values.